hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutputHelper.java

```diff
@@ -620,4 +623,18 @@ static void sleepIgnoreInterrupt(int retry) {
     } catch (InterruptedException e) {
     }
   }
+
+  public static String getDataNodeInfo(Collection<DatanodeInfo> datanodeInfos) {
+    if (datanodeInfos.isEmpty()) return "[]";
```

Comment:
How about we simplify this by using `Collectors.joining`?. For example we could do:
```
public static String getDataNodeInfo(Collection<DatanodeInfo> datanodeInfos) {
    return datanodeInfos.stream()
            .map(datanodeInfo -> "(" + datanodeInfo.getHostName() + "/"
                    + datanodeInfo.getInfoAddr() + ":" + datanodeInfo.getInfoPort() + ")")
            .collect(Collectors.joining(",", "[", "]"));
}
```

Also good idea to add UTs for the utility function!

Reply:
> How about we simplify this by using `Collectors.joining`?. For example we could do:
> 
> ```
> public static String getDataNodeInfo(Collection<DatanodeInfo> datanodeInfos) {
>     return datanodeInfos.stream()
>             .map(datanodeInfo -> new StringBuilder().append("(")
>                     .append(datanodeInfo.getHostName()).append("/")
>                     .append(datanodeInfo.getInfoAddr()).append(":")
>                     .append(datanodeInfo.getInfoPort()).append(")").toString())
>             .collect(Collectors.joining(",", "[", "]"));
> }
> ```
> 
> Also good idea to add UTs for the utility function!

This is great 👍🏻